src/components/Offline.jsx

```diff
@@ -0,0 +1,24 @@
+import { useConnectionChange } from "../hooks";
+import { useEffect, useState } from "react";
+
+const Offline = () => {
+  const [isDisconnected, setIsDisconnected] = useState(true);
+
+  useEffect(() => {
+    setIsDisconnected(useConnectionChange());
+    window.addEventListener("online", useConnectionChange());
+    window.addEventListener("offline", useConnectionChange());
```

Comment:
You can use a cleanup function for `useEffect` to removeEventListener for both listeners